pkg/codegen/nodejs/gen_test.go

```diff
@@ -17,34 +15,57 @@ import (
 )
 
 func TestGeneratePackage(t *testing.T) {
-	test.TestSDKCodegen(t, "nodejs", GeneratePackage, typeCheckGeneratedPackage)
+	test.TestSDKCodegen(t, &test.SDKCodegenOptions{
+		Language:   "nodejs",
+		GenPackage: GeneratePackage,
+		Checks: map[string]test.CodegenCheck{
+			"nodejs/compile": typeCheckGeneratedPackage,
+		},
+	})
 }
 
 func typeCheckGeneratedPackage(t *testing.T, pwd string) {
 	var err error
-	var npm string
-	npm, err = executable.FindExecutable("npm")
-	require.NoError(t, err)
-
 	var stdout, stderr bytes.Buffer
 	cmdOptions := integration.ProgramTestOptions{
 		Verbose: true,
 		Stderr:  &stderr,
 		Stdout:  &stdout,
 	}
 
-	// TODO remove when https://github.com/pulumi/pulumi/pull/7938 lands
-	file := filepath.Join(pwd, "package.json")
-	oldFile, err := ioutil.ReadFile(file)
+	// TODO: previous attempt used npm. It may be more popular and
```

Comment:
Do we want to support compiling our codegen output with `npm`? 
I'm thinking about the experience for authoring multi-lang components. If we say that `yarn` is required, then this decision makes sense. If we say you can also use `npm`, then I think we should test the more brittle system. I genuinely don't know our policy. Thoughts?

Reply:
Oh you've mentioned that, it's a compelling argument I'm just waiting on #7938 to work and then also we need to figure out how to make NPM link against the in-source SDK (similar to yarn link), in the meanwhile the yarn change gets this rolling. I can create an issue to remember to switch back?

Reply:
I'll make a note of it in #7938, I don't think it needs a separate issue. 

> then also we need to figure out how to make NPM link against the in-source SDK (similar to yarn link)

`npm` also has a [link feature](https://docs.npmjs.com/cli/v7/commands/npm-link). I assume we would just use that. 

I'm happy to merge this part as is, and fix it later.